server/src/main/kotlin/fi/oph/kitu/yki/YkiService.kt

```diff
@@ -16,20 +18,22 @@ class YkiService(
 ) {
     private val logger: Logger = LoggerFactory.getLogger(javaClass)
 
-    fun importYkiSuoritukset() {
-        val suoritus =
-            ykiRestClient
-                .get()
-                .uri("/yki")
-                .retrieve()
-                .body<YkiSuoritus>()
+    fun importYkiSuoritukset(lastSeen: LocalDate? = null) {
+        val request = ykiRestClient.get().uri("suoritukset")
+        val isDryRun = request.isDryRun()
+        val suoritukset = request.retrieve().csvBody<YkiSuoritus>()
 
-        if (suoritus == null) {
-            logger.info("No YKI suoritus found")
+        if (suoritukset.isEmpty()) {
+            logger.error("YKI responded with empty data.")
+            throw RestClientException("The response is empty")
+        }
+
+        if (isDryRun) {
+            logger.info("dry run complete")
             return
         }
 
-        repository.insertSuoritus(suoritus)
-        logger.info("YKI Suoritus added to repository")
+        repository.insertSuoritukset(suoritukset)
+        logger.info("suoritukset was added.")
```

Comment:
Tässä voisi lisätä lokitietueeseen esim. `count: suoritukset.size`
```kotlin
if (isDryRun) {
    logger.info("dry run complete")
    return
}

repository.insertSuoritukset(suoritukset)
logger
    .atInfo()
    .addKeyValue("count", suoritukset.size)
    .log("suoritukset was added.")
```
Lisäksi vaihtoehto olisi yhdistää myös dry-run lokirivi samalle riville, ja lisätä myös `dry-run` arvo lokitietueeseen, näin meillä ei olisi erillistä ehtolausetta ihan vain lokien muotoilua varten, vaan ehdon merkitys toiminnallisuuteen olisi hieman selkeämmin esillä:
```kotlin
if (!isDryRun) {
    repository.insertSuoritukset(suoritukset)
}

logger
    .atInfo()
    .addKeyValue("dry-run", isDryRun)
    .addKeyValue("count", suoritukset.size)
    .log("done importing suoritukset")
```